src/fides/api/ops/models/privacy_preference.py

```diff
@@ -110,6 +115,29 @@ class PrivacyPreferenceHistory(Base):
         uselist=False,
     )
 
+    def cache_system_status(self, system: str, status: ExecutionLogStatus) -> None:
+        """Update the cached affected system status for consent reporting
+
+        Typically this should just be called for consent connectors only.
+        If no request is made or email is sent, this should be called with a status of skipped.
+        """
+        db = Session.object_session(self)
+        if not self.affected_system_status:
+            self.affected_system_status = {}
+        self.affected_system_status[system] = status.value
```

Comment:
Is `system` here the equivalent to `System.key`?

Reply:
well, ConnectionConfig.system_key, it seemed like the requirements wanted to know things at the system level, but since connectors don't always have a system, I fall back to the connection config name if it's an orphan.